src/python/turicreate/toolkits/style_transfer/_tf_model_architecture.py

```diff
@@ -24,32 +24,37 @@ def define_tensorflow_variables(net_params, trainable=True):
     Parameters
     ----------
     trainable: boolean
-        If `true` the network updates the convolutional layers as well as the
-        instance norm layers of the network. If `false` only the instance norm
-        layers of the network are updated.
+        If `true` the transformer network updates the convolutional layers as
+        well as the instance norm layers of the network. If `false` only the
+        instance norm layers of the network are updated.
+
+        Note the VGG network's parameters aren't updated
     Returns
     -------
     out: dict
         The TF Variable dictionary.
     """
     tensorflow_variables = dict()
     for key in net_params.keys():
-        if "weight" in key:
-            if "conv" in key:
+        if 'weight' in key:
+            # only set the parameter to train if in the transformer network
+            train_param = trainable and 'transformer_' in key
+            if 'conv' in key:
                 tensorflow_variables[key] = _tf.Variable(
                     initial_value=_utils.convert_conv2d_coreml_to_tf(net_params[key]),
                     name=key,
-                    trainable=trainable,
+                    trainable=train_param,
```

Comment:
What is the transformer layer? Why is trainable switching between True and False?

Reply:
So trainable specifies if the convolutional layers of the `transformer` network are trainable or not. The instance normalization layers are always trainable whereas the convolutional layers of the `vgg` network should never be trainable. Since the `vgg` network only contains convolutional layers. and the only other network that contains convolutional layers is the `transformer` network, this split works.